Add instructions for running the unit tests in README

vladimir-v-diaz opened the pull request:
I encountered unit tests failures while trying to run the unit tests.

Without instructions, my first attempt was to navigate to the `uptane/tests` directory and run `./runtests.py`.  This resulted in unit test failures.  I wasn't sure if these were expected failures, or not. 
 The README does not currently provide instructions on running tests.

We should add instructions for running the tests so that potential collaborators do not run into the same issue.

vladimir-v-diaz committed:
Add instructions for running the unit tests in README

awwad commented:
Yes, that is expected in that they were run from the wrong directory. Tests are run from the root uptane/ directory. This is somewhat apparent from the text of runtests.py. Also, if you just run tox from the root uptane/ directory, all this is run correctly.

I'm not opposed to sticking instructions on how to run the tests in the README somewhere, (which would probably just tell people to run tox), and I'm not opposed to adding a small note about how to run individual tests (which would tell you to run them from the root directory, e.g. python tests/test_secondary.py). At the moment, though, the tests are a development tool, and the README is a set of instructions for running the demo to acquaint yourself with Uptane.

vladimir-v-diaz committed:
Edit Testing section to mention that individual tests can be executed

vladimir-v-diaz commented:
I added a commit to mention that individual unit tests can also be executed from the root directory, and that `runtests.py` runs all of the unit tests...

awwad committed:
DOC: Edit README notes on unit/integration tests

to get them out of the way a bit (so that they don't look like the
(confusing) default thing you should do when trying out Uptane).
Also explain why they may be useful.

awwad commented:
I made some edits to move the testing notes out of the way and explain why testing may be useful (and also adjust the instructions for the runtests options). Let me know if the result strikes you as worse. Otherwise, LGTM with the edits.

vladimir-v-diaz commented:
The edits look good to me.  You can go ahead and merge.

Note that people who are implementing Uptane might be interested in these unit tests, to help them with verifying behavior.

awwad committed:
DOC: Expand on reasons to run unit/integration tests in README

awwad commented:
Good point. Added that note. Merging.
